packages/frontend/core/src/modules/explorer/views/tree/node.tsx

```diff
@@ -117,6 +120,27 @@ export const ExplorerTreeNode = ({
   const [renaming, setRenaming] = useState(defaultRenaming);
   const [lastInGroup, setLastInGroup] = useState(false);
   const rootRef = useRef<HTMLDivElement>(null);
+  const { emoji, name } = useMemo(() => {
+    if (!extractEmojiAsIcon || !rawName) {
+      return {
+        emoji: null,
+        name: rawName,
+      };
+    }
+    const isStartsWithEmoji = /^(\p{Emoji_Presentation})/u.test(rawName);
+    if (isStartsWithEmoji) {
+      // emoji like "👨🏻‍❤️‍💋‍👨🏻" are combined. Graphemer can handle these.
+      const emojiEnd = Graphemer.nextBreak(rawName, 0);
```

Comment:
Should be able to be replaced by Intl.Segmenter

Reply:
We have used Grapemer in the indexer already. `Intl.Segmenter` may have different results on different browsers. I prefer Grapemer.

Reply:
how about Graphemer's perf? i think the native solution may perform better 
https://arc.net/l/quote/ftyvfooe

Reply:
Some interesting discussion https://github.com/eslint/eslint/issues/17835

Reply:
Or you can use this library https://github.com/cometkim/unicode-segmenter.
One issue with Intl.Segmenter is that it doesn't work properly in the CI environment of nodejs because there is no intl data in CI.

Reply:
how does Graphemer deal with intl data in this case?